crates/agentgateway/src/llm/mod.rs

```diff
@@ -190,6 +190,8 @@ impl AIProvider {
 						uri.authority = Some(Authority::from_str(&provider.get_host())?);
 						Ok(())
 					})?;
+					// Store the region in request extensions so AWS signing can use it
+					req.extensions_mut().insert(provider.region.clone());
```

Comment:
Because extensions is storing a map based only on type, this is saying `String --> <some-region>`. This is pretty ambiguous. Its generally best to make a dedicated wrapper for it like `struct AwsRegion(String)` to make it un-ambiguous